backend/src/main/java/de/amos/apachepulsarui/service/TopicService.java

```diff
@@ -104,4 +112,32 @@ private String getOwnerBroker(String fullTopicName) {
         }
     }
 
+    public ProducerDto getProducerByTopic(String topic, String producer) {
+        TopicStats topicStats = getTopicStats(topic);
+        PublisherStats publisherStats = topicStats.getPublishers().stream()
+                .filter(ps -> Objects.equals(ps.getProducerName(), producer))
+                .findFirst().orElseThrow();
+        return create(publisherStats, getMessagesByProducer(topic, topicStats.getSubscriptions().keySet(), producer));
+    }
+
+
+    private List<MessageDto> getMessagesByProducer(String topic, Set<String> subscriptions, String producer) {
```

Comment:
I'm feared I didn't mention it in my voice messages, but we agreed with the customer that Messages should be "behind" subscriptions, like the initial plan we had. So when a user clicks on a subscription, then we will fetch the messages for this subscription.

Reply:
Sorry, i think i just go lost and missed that out, quickly added it:)